packages/actor-query-operation-group/lib/GroupsState.ts

```diff
@@ -120,15 +145,31 @@ export class GroupsState {
       const single: Record<string, Term> = {};
       for (const aggregate of this.pattern.aggregates) {
         const key = termToString(aggregate.variable);
-        const value = AggregateEvaluator.emptyValue(aggregate);
+        const value = AsyncAggregateEvaluator.emptyValue(aggregate);
         if (value !== undefined) {
           single[key] = value;
         }
       }
       rows = [ Bindings(single) ];
     }
+    this.waitResolver(rows);
+  }
 
-    return rows;
+  /**
+   * Collect the result of the final state. This returns a Bindings per group,
+   * and a (possibly empty) Bindings in case no Bindings have been consumed yet.
+   * you can only call this method once. Once the promise resolves calling @{consumeBindings} will not alter this value.
```

Comment:
> Once the promise resolves calling @{consumeBindings} will not alter this value.

Makes me wonder if we also want to throw in `consumeBindings` if `this.resultHasBeenCalled` is true.

Reply:
I think that would be a valid thing to do.